.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+on: [push, pull_request]
+name: build
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        ghc: ['8.10']
+        os: [ubuntu-latest]
+    name: Haskell GHC ${{ matrix.ghc }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1.2.9
+        id: setup-haskell-build
+        with:
+          ghc-version: ${{ matrix.ghc }} # Resolves to the latest point release of GHC 8.8
+          cabal-version: '3.6.2.0' # Exact version of Cabal
+          enable-stack: true
+          stack-version: '2.7'
+      - name: Install dependencies
+        run: sudo apt-get install -y libxml2-utils libgl1-mesa-dev libglu1-mesa-dev freeglut3-dev libpulse-dev libblas-dev liblapack-dev libasound2-dev
+      - name: Cabal update
+        run: cabal update
+      - name: Freeze Cabal plan
+        run: cabal freeze
```

Comment:
I wonder whether `cabal freeze` should instead be done manually, and the result committed to the repo.

Reply:
I don't have much experience lately with cabal, but I think that would make the builds more deterministic, making it more similar to stack in that respect, so I think it might be a good idea.

Reply:
Can you rebase on `master` and see whether it works without this step?